### PR TITLE
Selected Panel Part #1

### PR DESCRIPTION
**Pull request by erri120**
Part of #1685. This includes the styles and the concept of a "selected panel". This is still missing the behavior, that'll come in part 2.

https://github.com/user-attachments/assets/5e55808f-f916-46c9-b546-c5d92df24922

### CODE REVIEW

File: src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs [hunk 1]
```diff
@@ -32,6 +34,19 @@ public PanelView()
                 .Subscribe()
                 .DisposeWith(disposables);
 
+            // panel selection
+            this.AddDisposableHandler(PointerEnteredEvent, (_, _) =>
+            {
+                if (ViewModel is not null) ViewModel.IsSelected = true;
+            }, routes: RoutingStrategies.Direct | RoutingStrategies.Bubble, handledEventsToo: true).DisposeWith(disposables);
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:41]: This changes selection on pointer entered hover, the desired behavior is selection to change on interaction with contents.

File: src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs [hunk 2]
```diff
@@ -32,6 +34,19 @@ public PanelView()
...
+
+            this.WhenAnyValue(view => view.IsKeyboardFocusWithin)
+                .Where(isFocused => isFocused)
+                .SubscribeWithErrorLogging(_ =>
+                {
+                    if (ViewModel is not null) ViewModel.IsSelected = true;
+                }).DisposeWith(disposables);
```
Comment (Al12rs) [inline src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: This does not seem to be doing anything from my testing

File: src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs (hunk 1 above)
Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:41]: Right, forgot to change that.

File: src/Themes/NexusMods.Themes.NexusFluentDark/Styles/UserControls/WorkspaceSystem/PanelViewStyles.axaml [hunk 3]
```diff
@@ -45,14 +66,8 @@
...
 
-                    <!-- selected tab -->
-                    <Style Selector="^ Border#Container.Selected">
-                        <Setter Property="BorderThickness" Value="1 0 0 0" />
-                    </Style>
-
-                    <!-- not selected tab -->
-                    <Style Selector="^ Border#Container:not(.Selected)">
-                        <Setter Property="BorderThickness" Value="1 0 0 1" />
+                    <Style Selector="^ Border#Container">
```
Comment (Al12rs) [inline src/Themes/NexusMods.Themes.NexusFluentDark/Styles/UserControls/WorkspaceSystem/PanelViewStyles.axaml:69]: Apparently this targets all Borders called "Container", even from child controls.
It seems we need to be more specific with with our naming or use more specific pathing

File: src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs (hunk 2 above)
Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: I tried using tab after disabling the pointer over version. I will try again if it is working on your end

Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: Ah, I found the problem. If you close the selected tab or open a new Workspace without any selection, it tabbing will not select the panels. If one panel is already selected, tabbing does select panels correctly.

Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: Actually pointer press doesn't seem to work either for new workspaces or if you close the selected panel

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: > Actually pointer press doesn't seem to work either for new workspaces or if you close the selected panel

Can you write down in detail how to repro? If you close a selected panel, then the workspace should select the first panel you have.

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: Closing selected panel, switching workspaces, and everything else works on my end.

Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: I tried cleaning my datamodel folder but now I can't get any panel to be selected after app start. No panel is selected and nothing selects panels

Comment (Al12rs) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: Ah, I got it. If you have a single panel there is no selection highlighting, which was confusing me.
Is that by design?

Comment (erri120) [reply src/NexusMods.App.UI/WorkspaceSystem/Panel/PanelView.axaml.cs:48]: Yeah

**Review by Al12rs — CHANGES_REQUESTED** [review]
Closing and reopening the app while one of the panels contains a NewTab page causes exceptions on reopening and a workspace without panels.

**Comment by Al12rs** [comment]
@captainsandypants in my testing of the feature I found the absence of the selected styling in the case of single panel very confusing. I was assuming a bug. 
The absence of the selected styling made me think something else must have been selected somewhere else or that the app was not responding correctly to interactions, as it does in multi panel scenarios.

**Review by Al12rs — APPROVED** [review]
Exception will be addressed in separate PR.

**Comment by captainsandypants** [comment]
> I found the absence of the selected styling in the case of single panel very confusing. I was assuming a bug. The absence of the selected styling made me think something else must have been selected somewhere else or that the app was not responding correctly to interactions, as it does in multi panel scenarios.

@Al12rs I find it intuitive as there's only one panel so it has to be the selected one. I want to avoid unnecessary UI wherever possible. We'll keep an eye on further feedback on this though.

**Comment by captainsandypants** [comment]
@erri120 when there's a single tab the UI doesn't match the designs. The right border shouldn't be edge to edge. See design here: 
image

**Comment by erri120** [comment]
> @erri120 when there's a single tab the UI doesn't match the designs. The right border shouldn't be edge to edge.

This was a compromise I made when implementing the original design for tabs months ago. The line is edge to edge because I set the border on the right to be 1px. There might be a way to change the geometry of the border using a custom brush, but that'll take a lot of experimenting to figure out. Another option would be to have no border, and just put a "line" element in between the tab and the plus icon. That would also take more time than it's worth, as that would require code-behind changes.

I couldn't justify spending hours on turning this border into a floating line, so we have this for now.

**Comment by erri120** [comment]
Added the separator

**Review by captainsandypants — APPROVED** [review]
All good.

**Pull request merged**